packages/babel-plugin-transform-typescript/src/index.js

```diff
@@ -134,7 +134,9 @@ export default declare(
               );
             }
 
-            return template.statement.ast`this.${id} = ${id}`;
+            return template.statement.ast`this.${t.cloneNode(
+              id,
+            )} = ${t.cloneNode(id)}`;
```

Comment:
Nit: I really dislike how this is formatted. Does prettier accept something like this?

```suggestion
            return template.statement.ast`
              this.${t.cloneNode(id} = ${t.cloneNode(id)}
            `;
```